src/ShallowWrapper.js

```diff
@@ -1078,13 +1100,14 @@ class ShallowWrapper {
   dive(options = {}) {
     const name = 'dive';
     return this.single(name, (n) => {
-      if (isDOMComponentElement(n)) {
+      if (n && n.nodeType === 'host') {
         throw new TypeError(`ShallowWrapper::${name}() can not be called on DOM components`);
```

Comment:
should this phrase - "DOM components" - come from the adapter, and not from enzyme itself?